ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReportingPage.tsx

```diff
@@ -24,14 +28,19 @@ function VulnReportingPage() {
 
     return (
         <Switch>
-            <Route exact path={vulnerabilityReportingPath} component={VulnReportsPage} />
-            {canReadWriteReports && (
-                <Route
-                    exact
-                    path={vulnerabilityReportingCreatePath}
-                    component={CreateVulnReportPage}
-                />
-            )}
+            <Route
```

Comment:
We're attempting to render components differently here. I wanted to keep all the routing logic in one place to keep things centralized and easy to navigate. We will use `render props` (old school 😎 ) to determine what component to render based on the route path and page actions.

We will have two routes here: `/main/vulnerabilities/reports` and `/main/vulnerabilities/reports/:reportId`

For `/main/vulnerabilities/reports`, we will have conditional logic to determine when to render the `CreateVulnReportPage` and when to render the `VulnReportsPage` (reports table). If the page action doesn't match these two routes, we will just render the `VulnReportsPage` page by redirecting to that path. Similar to how we do it in the other features

For `/main/vulnerabilities/reports/:reportId`, we will have conditional logic to determine when to render the `ViewReportPage`, `EditReportPage`, and `CloneReportPage`. If the page action doesn't match these two routes, we will just render the `ViewReportPage` page by redirecting to that path. Similar to how we do it in the other features

Reply:
I like this concept in general, especially when it comes to keeping the top level page components rendered at the same level. I agree that this will keep things easier to navigate.

In the hope of keeping things forward compatible, I was taking a look at react-router v6 to see if this is something we could replicate after an upgrade. (The "old school" comment made me wonder 😄 ) They don't have a way to do render props from what I can tell, but we might be able to replicate the same with the `Route.element` prop on the [new component](https://reactrouter.com/en/main/route/route#elementcomponent).

I think this could be a good place to load data for the page that we know we will always need when the page renders. React-router v6 seems to have a similar concept too but I haven't messed around with it.